server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -129,14 +151,21 @@ private RepositoryData(long genId, Map<String, SnapshotId> snapshotIds, Map<Stri
         this.indexMetaDataGenerations = indexMetaDataGenerations;
         this.snapshotVersions = snapshotVersions;
         assert indices.values().containsAll(shardGenerations.indices()) : "ShardGenerations contained indices "
-            + shardGenerations.indices() + " but snapshots only reference indices " + indices.values();
+                + shardGenerations.indices() + " but snapshots only reference indices " + indices.values();
         assert indexSnapshots.values().stream().noneMatch(snapshotIdList -> Set.copyOf(snapshotIdList).size() != snapshotIdList.size()) :
                 "Found duplicate snapshot ids per index in [" + indexSnapshots + "]";
     }
 
     protected RepositoryData copy() {
         return new RepositoryData(
-            genId, snapshotIds, snapshotStates, snapshotVersions, indexSnapshots, shardGenerations, indexMetaDataGenerations);
+                genId,
+                snapshotIds,
+                snapshotStates,
+                snapshotVersions,
+                indices,
```

Comment:
Adding `indices` so that this uses the private constructor to avoid re-wrapping everything in another `Collections.unmodifiableMap()`.